src/tools/schema2base.ts

```diff
@@ -29,18 +30,19 @@ export abstract class Schema2Base {
   }
 
   private async processFile(src: string) {
+    Utils.init(`${process.cwd()}/${src}`);
     const outName = this.opts.outfile || this.outputName(path.basename(src));
     const outPath = path.join(this.opts.outdir, outName);
     console.log(outPath);
     if (this.opts.update && fs.existsSync(outPath) && fs.statSync(outPath).mtimeMs > fs.statSync(src).mtimeMs) {
       return;
     }
 
-    const contents = fs.readFileSync(src, 'utf-8');
-    const manifest = await Manifest.parse(contents);
+    const manifest = await Utils.parse(`import '${process.cwd()}/${src}'`);
 
     // Collect declared schemas along with any inlined in particle connections.
-    const schemas: Dictionary<Schema> = {...manifest.schemas};
+    const schemas: Dictionary<Schema> = {};
+    manifest.allSchemas.forEach(schema => schemas[schema.name] = schema);    
```

Comment:
Are these names unique enough to ensure they don't accidentally overwrite if one has same name as another?

Do we have tests for `schema2base`, is it worth starting some to cover the new logic at minimum?

Reply:
This is a good point. I believe there could be multiple schemas with the same name. We should probably ask runtime team, or at least give them a heads-up .

Reply:
This is something we haven't yet addressed. Using schema names like this was a temporary solution to allow me to move forward on wasm. There are various issues with it: inline schemas can be un-named (`* {Text foo}`), schemas can have multiple names, schema names are not guaranteed unique, etc.

Possible solutions: name the classes based on the particle/handle combination, or require the user to define the names (maybe with the schema name as a fallback if available).